qualtran/bloqs/mod_arithmetic/mod_addition.py

```diff
@@ -115,7 +116,8 @@ def build_composite_bloq(self, bb: 'BloqBuilder', x: Soquet, y: Soquet) -> Dict[
         x = bb.join(x_split[1:], dtype=QMontgomeryUInt(bitsize=self.bitsize))
 
         # Add constant -p to the y register.
-        y = bb.add(AddK(bitsize=self.bitsize + 1, k=-1 * self.mod, signed=True, cvs=()), x=y)
+        # FIXME: signed addition not large enough to fit `self.bitsize`-ed number.
```

Comment:
this addition is unsiged ... it doesn't need fixing ... the flag `sigend=True` was wrong and you just fixed it... AddK will do the right thing  for `-p`

https://github.com/quantumlib/Qualtran/blob/0f4deeea16560b988c703011c2f44dadabcd1640/qualtran/bloqs/arithmetic/addition.py#L462-L467